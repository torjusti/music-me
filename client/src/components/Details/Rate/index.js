import React from 'react';
import classNames from 'class-names';
import { Icon } from 'semantic-ui-react';
import styles from './rate.module.css';

/**
 * A rating box consisting of checked and unchecked stars, which is
 * used to rate songs.
 */
const Rate = ({ onClick, rating = 0, maxRating = 5, ...rest }) => (
  <span {...rest}>
    {Array.from(Array(maxRating)).map((_, i) => (
      <Icon
        key={i}
        name={classNames('star', {
          outline: i >= rating,
        })}
        onClick={() => onClick(i + 1)}
        className={styles.star}
      />
    ))}
  </span>
);

export default Rate;
