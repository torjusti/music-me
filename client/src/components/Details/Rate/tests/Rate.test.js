import React from 'react';
import { shallow } from 'enzyme';
import Rate from '../index';
import configureMockStore from 'redux-mock-store'

describe('<Rate />', () => {

  // Create the mock store
  const mockStore = configureMockStore();

  let wrapper, store;

  beforeEach(() => {
    store = mockStore();
    // Shallow render the container passing in the mock store
    wrapper = shallow(
      <Rate store={store} />
    );
  });

  it('should render Rate snapshot correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Rate correctly', () => {
    expect(wrapper.length).toEqual(1);
  });
});
