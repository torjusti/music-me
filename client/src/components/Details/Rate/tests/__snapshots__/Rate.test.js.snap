// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rate /> should render Rate snapshot correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rate
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          as="i"
          className="star"
          name="star outline"
          onClick={[Function]}
        />,
        <Icon
          as="i"
          className="star"
          name="star outline"
          onClick={[Function]}
        />,
        <Icon
          as="i"
          className="star"
          name="star outline"
          onClick={[Function]}
        />,
        <Icon
          as="i"
          className="star"
          name="star outline"
          onClick={[Function]}
        />,
        <Icon
          as="i"
          className="star"
          name="star outline"
          onClick={[Function]}
        />,
      ],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "className": "star",
          "name": "star outline",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "className": "star",
          "name": "star outline",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "className": "star",
          "name": "star outline",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "className": "star",
          "name": "star outline",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "className": "star",
          "name": "star outline",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            as="i"
            className="star"
            name="star outline"
            onClick={[Function]}
          />,
          <Icon
            as="i"
            className="star"
            name="star outline"
            onClick={[Function]}
          />,
          <Icon
            as="i"
            className="star"
            name="star outline"
            onClick={[Function]}
          />,
          <Icon
            as="i"
            className="star"
            name="star outline"
            onClick={[Function]}
          />,
          <Icon
            as="i"
            className="star"
            name="star outline"
            onClick={[Function]}
          />,
        ],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "className": "star",
            "name": "star outline",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "className": "star",
            "name": "star outline",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "className": "star",
            "name": "star outline",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "className": "star",
            "name": "star outline",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "className": "star",
            "name": "star outline",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
